compiler/next/lib/parsing/parsing-queries.cpp

```diff
@@ -55,18 +56,35 @@ const FileContents& fileText(Context* context, UniqueString path) {
   return QUERY_END(result);
 }
 
-void setFileText(Context* context, UniqueString path, FileContents result) {
-  QUERY_STORE_INPUT_RESULT(fileText, context, result, path);
+const FileContents& fileText(Context* context, std::string path) {
+  return fileTextQuery(context, path);
+}
+
+const FileContents& fileText(Context* context, UniqueString path) {
+  return fileText(context, path.str());
+}
+
+void setFileText(Context* context, std::string path, FileContents result) {
+  QUERY_STORE_INPUT_RESULT(fileTextQuery, context, result, path);
+}
+void setFileText(Context* context, std::string path, std::string text) {
+  setFileText(context, std::move(path), FileContents(std::move(text)));
 }
 void setFileText(Context* context, UniqueString path, std::string text) {
-  setFileText(context, path, FileContents(std::move(text)));
+  setFileText(context, path.str(), FileContents(std::move(text)));
+}
+
+bool hasFileText(Context* context, const std::string& path) {
+  auto tupleOfArgs = std::make_tuple(path);
+
+  return context->hasCurrentResultForQuery(fileTextQuery, tupleOfArgs);
 }
 
 const uast::BuilderResult& parseFile(Context* context, UniqueString path) {
   QUERY_BEGIN(parseFile, context, path);
 
   // Run the fileText query to get the file contents
-  const FileContents& contents = fileText(context, path);
+  const FileContents& contents = fileText(context, path.str());
```

Comment:
Why don't we have an overload of `fileText` that just takes a `UniqueString` for the path?

Reply:
We do -- https://github.com/chapel-lang/chapel/pull/19134/files#diff-cade051bb37024d85ae3738912fc07abc0b338450080f7b18a05d2364d50d640R49 . I can make it call it here.